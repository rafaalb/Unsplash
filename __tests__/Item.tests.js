import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  friends: ['Emily', 'Rafael'],
  thumbnail: 'image.jpg',
  age: 20,
  weight: 200,
  height: 150,
  hair_color: 'red',
  professions: ['Carpenter', 'Computer Scientist'],
  name: 'Zedkin Thor'
};

describe('<Item />', () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);
  it('renders the image properly', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toEqual(fakeItem.thumbnail);
  });
  it('renders age and title', () => {
    const title = wrapper.find('Title a').text();
    const age = wrapper.find('Age__AgeTag');
    expect(age.children().text()).toEqual('20 years old');
    expect(title).toEqual('Zedkin Thor');
  });
  it('renders and matches the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
