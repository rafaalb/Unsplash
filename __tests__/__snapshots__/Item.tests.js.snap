// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    src="image.jpg"
  />
  <Title>
    <Link>
      <a>
        Zedkin Thor
      </a>
    </Link>
  </Title>
  <Item__ContentContainer>
    <WithStyles(ItemContainer)
      item={
        Object {
          "age": 20,
          "friends": Array [
            "Emily",
            "Rafael",
          ],
          "hair_color": "red",
          "height": 150,
          "name": "Zedkin Thor",
          "professions": Array [
            "Carpenter",
            "Computer Scientist",
          ],
          "thumbnail": "image.jpg",
          "weight": 200,
        }
      }
    />
    <WithStyles(FriendsList)
      item={
        Object {
          "age": 20,
          "friends": Array [
            "Emily",
            "Rafael",
          ],
          "hair_color": "red",
          "height": 150,
          "name": "Zedkin Thor",
          "professions": Array [
            "Carpenter",
            "Computer Scientist",
          ],
          "thumbnail": "image.jpg",
          "weight": 200,
        }
      }
    />
    <Age__AgeTag>
      20 years old
    </Age__AgeTag>
  </Item__ContentContainer>
</ItemStyles__Item>
`;
